src/plugins/procmon/linux.cpp

```diff
@@ -498,15 +498,15 @@ static event_response_t send_signal_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* i
     // Save data about current process
     lw->procmon = procmon;
     lw->process_name = current_proc_data.name;
-    lw->thread_name = current_thread_name;
+    lw->thread_name = current_thread_name ?: "";
     lw->pid = current_proc_data.pid;
     lw->tid = current_proc_data.tid;
     lw->ppid = current_proc_data.ppid;
     lw->rsp = ret_addr;
 
     // Save data about target process
     lw->target_process_name = target_proc_data.name;
-    lw->target_thread_name = target_thread_name;
+    lw->target_thread_name = target_thread_name ?: "";
```

Comment:
target_thread_name gets freed below, shouldn't this be some strdup type copy here?

Reply:
`lw->target_thread_name` is a `std::string`. Assignment makes implicit data copying.